src/coreclr/jit/fgopt.cpp

```diff
@@ -317,9 +317,23 @@ PhaseStatus Compiler::fgPostImportationCleanup()
             {
                 LclVarDsc* returnSpillVarDsc = lvaGetDesc(lvaInlineeReturnSpillTemp);
 
-                if ((returnSpillVarDsc->lvType == TYP_REF) && returnSpillVarDsc->lvSingleDef)
+                if ((returnSpillVarDsc->lvType == TYP_REF))
                 {
-                    lvaUpdateClass(lvaInlineeReturnSpillTemp, retExprClassHnd, impInlineInfo->retExprClassHndIsExact);
+                    if (returnSpillVarDsc->lvSingleDef)
+                    {
+                        lvaUpdateClass(lvaInlineeReturnSpillTemp, retExprClassHnd,
+                                       impInlineInfo->retExprClassHndIsExact);
+                    }
+                    else if (lvaInlineeReturnSpillTempFreshlyCreated)
+                    {
+                        // If the inlinee return spill temp was freshly created in fgFindBasicBlocks, we can update the
+                        // type regardless of whether it is single def or not.
+                        returnSpillVarDsc->lvClassHnd     = retExprClassHnd;
+                        returnSpillVarDsc->lvClassIsExact = impInlineInfo->retExprClassHndIsExact;
+#if DEBUG
+                        returnSpillVarDsc->lvClassInfoUpdated = true;
+#endif
+                    }
```

Comment:
```suggestion
                    if (returnSpillVarDsc->lvSingleDef || lvaInlineeReturnSpillTempFreshlyCreated)
                    {
                        lvaUpdateClass(lvaInlineeReturnSpillTemp, retExprClassHnd,
                                       impInlineInfo->retExprClassHndIsExact);
                    }
```
Can it be simplified like this?

Reply:
Perhaps just update the assertion to also allow updating the inlinee spill temp's class.